rs/execution_environment/src/history.rs

```diff
@@ -214,27 +199,30 @@ impl IngressHistoryWriter for IngressHistoryWriterImpl {
                     state: Received, ..
                 },
             ) => {
-                if let Some((seconds_since_block_made, _)) = message_latencies_seconds(&message_id)
-                {
-                    self.message_state_transition_received_duration_seconds
-                        .observe(seconds_since_block_made);
-                }
+                // Wall clock duration since the message was included into a block.
+                let duration_since_block_made = system_time_now().saturating_duration_since(time);
```

Comment:
This one line is the actual fix. Everything else is just refactoring and documentation.

Reply:
Just to make sure I understand: if we start using `system_time_now` (similar for the ones below), that means that the time observed can vary among different nodes. That is likely not the biggest issue as we would typically take the median or so when presenting the metrics but it does deviate a bit from other metrics produced by the DSM (and which are expected to be equal across all nodes).

I don't think this is a blocker to get this PR in but wanted to be sure that this is an explicit decision we're taking.

Reply:
You are correct.

The intent with this metric (and the other, already working, recently added ones) is to measure user perceived (induction / processing / ingress queue) latency. Relying exclusively on block times would only tell us the time deltas between blocks. FWIW, this is how that looks for `scheduler_canister_ingress_queue_latencies_seconds` (the per-canister mean time spent in an ingress queue) for each of the already updated subnets over the past day:

![Screenshot from 2024-11-19 13-38-13](https://github.com/user-attachments/assets/4d08bab6-b1ed-4184-ad6c-583c880acdd1)

Not particularly interesting.

As a side note, while all replicas will observe the exact same durations when based on block times, they will not observe them at the same time; we will not scrape each replica at the same time (but with a random `[0s..10s)` shift); and when a replica restarts or state syncs it may observe the same event multiple times (or not at all).

Reply:
For the record, here is the same measurement as above, except per-message (instead of per-canister) and based on wall times:

![Screenshot from 2024-11-19 13-44-15](https://github.com/user-attachments/assets/3bec7566-4bac-4f75-8583-e0704ba9152a)

And a similar measurement for time to first execution (including said first execution):

![Screenshot from 2024-11-19 13-44-24](https://github.com/user-attachments/assets/539b5e23-7bdb-46e4-bc3c-34fa554885d5)